### DISCUSSION TITLE
Vite alternative for loader:scss:additionalData

### DISCUSSION THREAD

**Discussion by larsalbrecht** [nuxt/nuxt#16148] [Questions] [answered]
In nuxt 2 I have a custom loader to manipulate (s)css (add, replace, etc).

What would be the correct alternative in nuxt3?

**Comment by TheAlexLichter** [comment]
See [the docs](https://nuxt.com/docs/getting-started/assets#example-2).

```js
export default defineNuxtConfig({
  vite: {
    css: {
      preprocessorOptions: {
        scss: {
          additionalData: '@use "@/assets/_colors.scss" as *;'
        }
      }
    }
  }
})

```

**Comment by larsalbrecht** [comment]
Hi, 
sorry for the misunderstanding. I have already found SCSS global supplement. 
I want to use a function like before in nuxt 2:
https://webpack.js.org/loaders/sass-loader/#sassoptions

```
{
  loaders: {
    scss: {
      additionalData:(content, loaderContext) => {
        const newContent = content;
        // magic things
        return newContent
      }
    }
  }
}
```

**Reply by TheAlexLichter** [reply]
Ohh, got it. I think you can use `additionalData` as a function already instead of a plain string. But there won't be a loader context

**Reply by larsalbrecht** [reply]
Oh yes, of course! Works like a charm. Thank you

**Comment by larsalbrecht** [comment, marked as answer]
As @manniL  mentioned, it also works over it.

There is even the context:
1st parameter: Content
2nd parameter: Context (file containing the content)

https://github.com/nuxt/framework/discussions/9614#discussioncomment-4351616

(unfortunately he only commented, so here again so there is a solution)
